result-watcher.bal

```diff
@@ -404,6 +422,9 @@ public isolated class ResultWatcher {
             self.resultEndpoint = resultEndpoint;
             self.httpClient = check new (self.resultEndpoint);
         }
+        lock {
+            resultWatcherRegistry.push(self);
```

Comment:
maybe move this outside of init into a helper function on module level that creates a watcher and adds it to the registry